src/optimizer/rule/LimitPushDownRule.cpp

```diff
@@ -0,0 +1,200 @@
+
+    bool gnFound = false;
+    const OptGroupExpr *gnExpr = nullptr;
+    for (auto dep : projExpr->dependencies()) {
+        for (auto expr : dep->groupExprs()) {
+            if (expr->node()->kind() == PlanNode::Kind::kGetNeighbors) {
+                gnFound = true;
+                gnExpr = expr;
+                break;
+            }
+        }
+    }
+
+    if (!gnFound) {
+        return std::make_pair(false, matched);
+    }
+
+    const auto gn = static_cast<const GetNeighbors *>(gnExpr->node());
+    if (limitRows >= gn->limit()) {
+        return std::make_pair(false, matched);
+    }
+    matched.push_back(projExpr);
+    matched.push_back(gnExpr);
+    return std::make_pair(true, matched);
+}
+
+Limit *LimitPushDownRule::cloneLimit(QueryContext *qctx,
+                                     const Limit *limit) const {
+    auto newLimit = Limit::make(qctx, nullptr, limit->offset(), limit->count());
+    newLimit->setInputVar(limit->inputVar());
+    newLimit->setOutputVar(limit->outputVar());
+    return newLimit;
+}
+
+Project *LimitPushDownRule::cloneProj(QueryContext *qctx,
+                                      const Project *proj) const {
+    auto cols = qctx->objPool()->add(new YieldColumns());
+    for (auto col : proj->columns()->columns()) {
+        cols->addColumn((col->clone()).release());
+    }
+
+    auto newProj = Project::make(qctx, nullptr, cols);
+    newProj->setInputVar(proj->inputVar());
+    newProj->setOutputVar(proj->outputVar());
+    return newProj;
+}
+
+GetNeighbors *LimitPushDownRule::cloneGetNbrs(QueryContext *qctx,
+                                              const GetNeighbors *gn) const {
+    auto newGn = GetNeighbors::make(qctx, nullptr, gn->space());
+    newGn->setSrc(gn->src());
+    newGn->setEdgeTypes(gn->edgeTypes());
+    newGn->setEdgeDirection(gn->edgeDirection());
+    newGn->setDedup(gn->dedup());
+    newGn->setRandom(gn->random());
+    newGn->setFilter(gn->filter());
+    newGn->setLimit(gn->limit());
+    newGn->setInputVar(gn->inputVar());
+    newGn->setOutputVar(gn->outputVar());
+
+    if (gn->vertexProps()) {
+        auto vertexProps = *gn->vertexProps();
+        auto vertexPropsPtr = std::make_unique<decltype(vertexProps)>(std::move(vertexProps));
+        newGn->setVertexProps(std::move(vertexPropsPtr));
+    }
+
+    if (gn->edgeProps()) {
+        auto edgeProps = *gn->edgeProps();
+        auto edgePropsPtr = std::make_unique<decltype(edgeProps)>(std::move(edgeProps));
+        newGn->setEdgeProps(std::move(edgePropsPtr));
+    }
+
+    if (gn->statProps()) {
+        auto statProps = *gn->statProps();
+        auto statPropsPtr = std::make_unique<decltype(statProps)>(std::move(statProps));
+        newGn->setStatProps(std::move(statPropsPtr));
+    }
+
+    if (gn->exprs()) {
+        auto exprs = *gn->exprs();
+        auto exprsPtr = std::make_unique<decltype(exprs)>(std::move(exprs));
+        newGn->setExprs(std::move(exprsPtr));
+    }
+    return newGn;
+}
```

Comment:
Emm, these `clone`s should be provided by the node itself.

Reply:
this is fixed in yee's pr https://github.com/vesoft-inc/nebula-graph/pull/311/files